cmd/participation/list.go

```diff
@@ -0,0 +1,49 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package main
+
+import (
+	"fmt"
+	"os"
+)
+
+// Lists the channels an OSN is a member of.
+func listAllChannels(osn, tlsDir string) {
+	url := fmt.Sprintf("https://%s/participation/v1/channels", osn)
+
+	client := httpClient(tlsDir)
+	bodyBytes, err := getBodyBytes(client, url)
+	if err != nil {
+		panic(err)
+	}
+
+	err = printResponseAsJSON(bodyBytes, &channelList{}, os.Stdout)
+	if err != nil {
+		panic(err)
+	}
+}
+
+// Lists a single channel an OSN is a member of.
+func listSingleChannel(osn, tlsDir, channelID string) {
+	url := fmt.Sprintf("https://%s/participation/v1/channels/%s", osn, channelID)
+
+	client := httpClient(tlsDir)
+	bodyBytes, err := getBodyBytes(client, url)
+	if err != nil {
+		panic(err)
+	}
+
+	if bodyBytes == nil {
+		fmt.Printf("channel '%s' not found\n", channelID)
+		return
+	}
```

Comment:
There is a specific error code that captures this. I think it is better to follow these.

Reply:
Refactored. Done.